### ISSUE TITLE
Salt/Chlorine deficit

### ISSUE THREAD

**Issue by neoncyber1** [AztechMC/Modern-Industrialization#989]
So, salt is used to make chlorine, which is used to make polyvinyl, which is ultimately needed for titanium drills and upgrades. The most efficient way to make polyvinyl is to use chromium (and it is also required to make stainless steel btw), which also requires chlorine. So to maintain the high titanium drill quarry mining speed, I need to make more extra quarries simply due to the lack of just salt to keep enough resources. I think they way we get chlorine from salt (0,125b per 2 dusts) is way too inefficient and it might have to be doubled.

**Comment by Swedz** [comment]
Are you recycling your chlorine from processing chromium? You tend to get more salt than you will use from one quarry as long as you're properly recycling your chlorine.

**Comment by neoncyber1** [comment]
> Are you recycling your chlorine from processing chromium? You tend to get more salt than you will use from one quarry as long as you're properly recycling your chlorine.

Yes i do but 1 chromium cycle gives 0.45b less chlorine and eventually i run out of salt. I got 2 max bronze-drill quarries for redstone->chromium and 1 max steel-drill quarry for salt and my hydrochloric acid generation is way to low to provide enough polyvinyl for titanium drills.
![image](https://github.com/user-attachments/assets/317629ec-1593-4d63-acf5-84aea336f861)
![image](https://github.com/user-attachments/assets/c1cde444-7830-445e-bc35-24740c3524b9)

**Comment by neoncyber1** [comment]
Yes i do but 1 cycle gives 0.45b chlorine less and eventually i dont recieve enough salt for chromium cycle and titanium drills according to my data if its correct:
![image](https://github.com/user-attachments/assets/6a78a4da-ac3e-43cd-9fdd-d0926c6a6cfc)
![image](https://github.com/user-attachments/assets/4e09ac04-0d3d-4abd-9a1d-365bfd9717f1)
i got 2max overclocked bronze-drill quarries for redstone->chromium and 1max steel-drill for salt and my generation after chromium cycle is 0,02b/t of hydrochloric acid thats not enough for titanium drills

**Comment by truepurpledino** [comment]
how many output slots are you using on your quarries

from experience chlorine is never an issue on vanilla MI, if you use everything correctly

**Comment by neoncyber1** [comment]
Here's some clean data based on a 12% chance of getting salt ore per tick, and it shows ideal conditions (enough slot hatches) where I get all the ore and not affected by any ingame experience.

**Comment by neoncyber1** [comment]
> from experience chlorine is never an issue on vanilla MI, if you use everything correctly

I generate 3/t redstone = 0,05/t  chromium crushed dust or 1/s, one chrused dust cycle requires 4,5b chlorine and gives 4,05b clorine so i am losing 0,45b/s chlorine for chromium, max quarry gives 0,12/t salt ore = 0,54/t salt, chlorine recipe is 0,125 for 2 salt = 0,03375/t = 0,675/s of chlorine. So after losing chlorine for chromium i get 0,675-0,45=0,225/s chlorine. 1 titanium drill costs 0,402 chlorine to craft  and consuption chance on max overclock is 4%/t = 0,8/s drill = 322/s chlorine > than i generate and i also need smth left to craft. Chlorine deficit is the only  i experience through all processes.

**Comment by truepurpledino** [comment]
if you are using titanium drills at tick speed I don't know why you'd only still be running a single steel drill quarry at tick speed

Edit: I reviewed your numbers, they are correct. But I don't understand why you think this is a problem, since you can just make more steel quarries. Most players never come close to running titanium drills at tick speed because of the resources and energy costs, and it's usually not necessary in a normal, vanilla MI run (and in most modpacks that use MI).

**Comment by neoncyber1** [comment]
For the most part you are right. Just reporting that compared to all other resources salt/chlorine feels kinda deficit.

**Comment by Technici4n** [comment]
According to my math, to keep a titanium drill quarry going at 1 craft/tick, you need 0.0938/t chromium dust (chlorine cost: * 150 so 14 mb/t), and 16.1 mb/t additional chlorine for PVC. That's a little less than what **one** steel quarry can provide.
```
Chlorine production for 1 steel quarry:
0.12/t salt ore
0.36/t salt crushed dust
0.54/t salt dust
33.75 mb/t chlorine

Chromium production for 1 bronze quarry:
0.4/t redstone ore
1.8/t redstone dust
0.18/t ruby dust
0.03/t chromium crushed dust
0.09/t chromium dust
consumes 13.5 mb/t chlorine (for chromium)

4 titanium drills cost:
9.38 chromium dust
45000/28 mb chlorine

0.04/t titanium drill costs:
0.0938/t chromium dust
16.1 mb/t chlorine (on top of chromium)
```

Based on these numbers, you should have enough chlorine, assuming that most of it is used by the quarry. Let me know if you spot a mistake.

**Comment by warbaque** [comment]
Resources needed to run 1 titanium drill at tick speed: https://katiska.dy.fi/temp/minecraft/factoriolab/list?p=R-8168**3&s=mcmi**0&v=9

![Screenshot_2025-01-03-04-05-24-55_40deb401b9ffe8e1df2f1cc5ba480b12](https://github.com/user-attachments/assets/f11e06f3-fda9-4a7d-bb1f-fc05246425b2)

You need 0.13 salt ore per tick, your 3 steel drills produce 0.36 per tick -> you're left with excess of 0.23 per tick.

Salt is not the bottleneck from steel drills, bauxite is.

![Screenshot_2025-01-03-04-16-11-82_40deb401b9ffe8e1df2f1cc5ba480b12](https://github.com/user-attachments/assets/b0da1704-8f87-4c2b-a494-c91ea2e5731a)

83% of chlorine comes from hydrochloric solution
